model/flow/header.go

```diff
@@ -76,6 +77,13 @@ func (h Header) Body() interface{} {
 	}
 }
 
+func (h *Header) StructureValid() error {
```

Comment:
I think there are a lot of thinks that we could check here to cover structural validity of the header:
* `ParentVoterIndices`, `ParentVoterSigData`, `ProposerSigData` should all not be empty or nil
* `ParentID` and `ProposerID` should not be `flow.ZeroID`

We could check them here. Not sure if we want to from a design perspective. Also, there are exceptions for the genesis block 🤷. I think that would be worth-while documenting here.

Reply:
I omitted these exactly because of the root block exception that you mentioned.